src/EditorFeatures/Core/Implementation/Workspaces/EditorTextFactoryService.cs

```diff
@@ -94,7 +94,7 @@ private unsafe SourceText CreateTextFromTemporaryStorage(ISupportDirectMemoryAcc
             using (var reader = new DirectMemoryAccessStreamReader(src + 1, streamLength / sizeof(char) - 1))
             {
                 var buffer = CreateTextBuffer(reader, cancellationToken);
-                return buffer.CurrentSnapshot.AsRoslynText(Encoding.Unicode);
+                return buffer.CurrentSnapshot.AsRoslynText(encoding ?? Encoding.Unicode);
```

Comment:
@pharring 

What I'm having trouble understanding is the following

``` csharp
encoding ?? Encoding.Unicode
```

That implies it's possible to get here without an encoding.  In that case we replace it with an encoding which will add a BOM to a file if this `SourceText` is later written to disk.  That seems to go against most other scenarios where no BOM would be added.